test/api/capi/test_capi_complex_types.cpp

```diff
@@ -757,3 +756,83 @@ TEST_CASE("Array value construction") {
 	}
 	duckdb_destroy_value(&array_value);
 }
+
+TEST_CASE("Map value construction") {
```

Comment:
This is missing some unhappy path testing

Can you add tests with:
- `map_type` as not a MAP LogicalType
- `map_type` as null
- `keys_v` not matching the key type of the `map_type`
- `keys_v` as null
- `values_v` not matching the value type of the `map_type`
- `values_v` as null
- `keys_v` being constructed with differing types for the child values[1]
- `values_v` being constructed with differing types for the child values[1]

[1] Looking at the code, this doesn't seem to be detected and errored on by `duckdb_create_list_value`, which is kind of an oversight, it would be great if you can add tests for that, possibly fix `duckdb_create_list_value` in this PR by just checking that the provided values have the right type.